src/solvers/refinement/string_constraint_generator_main.cpp

```diff
@@ -600,7 +608,7 @@ string_exprt string_constraint_generatort::add_axioms_from_char_array(
 string_exprt string_constraint_generatort::add_axioms_from_char_array(
   const function_application_exprt &f)
 {
-  assert(false); // deprecated, we should use add_axioms_for_substring instead
+  UNREACHABLE; // deprecated, we should use add_axioms_for_substring instead
```

Comment:
If `UNREACHABLE` were to take a parameter, we could turn this comment into an error message.